state/protocol/events.go

```diff
@@ -84,10 +84,21 @@ type Consumer interface {
 	// NOTE: Only called once the phase transition has been finalized.
 	EpochCommittedPhaseStarted(currentEpochCounter uint64, first *flow.Header)
 
-	// EpochEmergencyFallbackTriggered is called when epoch fallback mode [EFM] is triggered.
-	// Since EFM is a permanent, spork-scoped state, this event is triggered only once.
-	// After this event is triggered, no further epoch transitions will occur,
-	// no further epoch phase transitions will occur, and no further epoch-related
-	// related protocol events (the events defined in this interface) will be emitted.
-	EpochEmergencyFallbackTriggered()
+	// EpochFallbackModeTriggered is called when epoch fallback mode [EFM] is triggered.
+	// EFM is triggered when an invalid or unexpected epoch-related service event is observed,
+	// or an expected service event is not observed before the epoch commitment deadline.
+	// After EFM is triggered, we remain in the current epoch until an EpochRecover
```

Comment:
I think it's not 100% accurate, we will enter next epoch even in EFM if it was previously committed.

Reply:
Good point, updated this doc to be more specific in https://github.com/onflow/flow-go/pull/6062/commits/5af958363ad8a9b6635fe283743c57fb028384e9